File: Easy/1. Two Sum/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++)
            map.put(target - nums[i], i);
        
        for(int j = 0; j < nums.length; j++)
            if(map.containsKey(nums[j])) {
                int i = map.get(nums[j]);
                if(i != j)
                    return new int[] { j, i };
            }

        return null;
    }
}

File: Easy/1. Two Sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
  let map = {};

  for (let i = 0; i < nums.length; ++i) {
    let comp = target - nums[i];

    if (map[comp] != undefined)
        return [map[comp], i];

    map[nums[i]] = i;
  }

  return undefined;
}
